pkg/controller/plan/adapter/converter.go

```diff
@@ -0,0 +1,259 @@
+package adapter
+
+import (
+	"context"
+	"fmt"
+
+	plancontext "github.com/konveyor/forklift-controller/pkg/controller/plan/context"
+	"github.com/konveyor/forklift-controller/pkg/controller/provider/web/base"
+	liberr "github.com/konveyor/forklift-controller/pkg/lib/error"
+	"github.com/konveyor/forklift-controller/pkg/lib/logging"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type FilterFn func(pvc *v1.PersistentVolumeClaim) bool
+
+type PVCConverter interface {
+	ConvertPVC(pvcs *v1.PersistentVolumeClaim, srcFormat, dstFormat string) (ready bool, err error)
```

Comment:
and following up on the other thread, then we won't need dstFormat at that level because each provider would have a different one (OSP - raw, vSphere - none, export OVA - qcow2/vmdk)

Reply:
actually not sure about the vSphere part